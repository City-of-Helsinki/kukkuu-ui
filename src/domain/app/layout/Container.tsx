import React, { FunctionComponent } from 'react';
import classnames from 'classnames';

import styles from './container.module.scss';

const stringStartsWith = (targetString: string, searchString: string) => {
  return targetString.substring(0, searchString.length) === searchString;
};

const Container: FunctionComponent<{
  children?: any;
  className?: string;
}> = ({ children, className }) => {
  const isHome = (
      children.props.className &&
      stringStartsWith(children.props.className, 'home_home_'));
  return (
    <div>
      {isHome ? (
        <div className={classnames(styles.container, className, styles.gridLayoutOverride)}>{children}</div>
      ) : (
        <div className={classnames(styles.container, className)}>{children}</div>
      )}
    </div>
  );
};

export default Container;
