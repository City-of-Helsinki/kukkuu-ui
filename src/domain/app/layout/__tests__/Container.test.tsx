import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Container from '../Container';

it('renders snapshot correctly', () => {
  const props = {}
  const container = shallow(<Container children={{props}} />);
  expect(toJson(container)).toMatchSnapshot();
});
