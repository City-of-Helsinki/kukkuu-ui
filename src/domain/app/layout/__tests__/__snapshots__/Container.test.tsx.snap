// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div>
  <div
    className="container"
  >
    <Component />
  </div>
</div>
`;
